### PR TITLE
Use ParameterValueWidget for default value picker in native parameter settings

### LINKED ISSUE #44665: "Dropdown list" input used for default filter widget value when "Search box" is chosen
### Describe the bug

https://github.com/metabase/metabase/assets/6830683/9e4e740a-9f8b-46b9-97af-5571ebe8499c

### To Reproduce

1. Start a new native question
2. Type `select {{param}}`
3. In the parameter sidebar choose "Dropdown list", then "Edit"
4. Provide few values for the "Custom list" and hit "Done"
5. Verify that dropdown component is used for both parameter value input (above query editor) and "Default filter widget value" (in the parameter sidebar)
6. Change "Dropdown list" to "Search box"

Search box is correctly used for parameter value input (above query editor) but dropdown component is used for "Default filter widget value" (sidebar).

### Expected behavior

The same component is used for default value input as for parameter value input

### Information about your Metabase installation

master, 70eea1167b

### Severity

P2

### PR DESCRIPTION
**Pull request by romeovs**
Closes #44665 

### To verify the changes

1. Start a new native question
2. Type `select {{param}}`
3. In the parameter sidebar choose "Dropdown list", then "Edit"
4. Provide few values for the "Custom list" and hit "Done"
5. Verify that dropdown component is used for both parameter value input (above query editor) and "Default filter widget value" (in the parameter sidebar)
6. Change "Dropdown list" to "Search box"

The same component should be used for the actual parameter and the default picker.

Note the the pickers for the search box and dropdown list currently look very similar. That is being fixed in: https://github.com/metabase/metabase/pull/49577

https://github.com/user-attachments/assets/922e1823-43fd-40d8-bc0e-22e82d67805e

### CODE REVIEW

File: e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js [hunk 1]
```diff
@@ -1546,3 +1548,46 @@ describe("issue 49321", () => {
...
+    modal().within(() => {
+      cy.findByText("Custom list").click();
+      cy.findByRole("textbox").type("foo\nbar\nbaz");
+      cy.button("Done").click();
+    });
+
+    sidebar().last().findByText("Enter a default value…").click();
+    popover()
+      .findByPlaceholderText("Enter a default value…")
+      .should("be.visible");
```
Comment (kamilmielnik) [inline e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js:1576]: Let's also assert that filter values are shown in the popover

```suggestion
    popover().within(() => {
      cy.findByPlaceholderText("Enter a default value…").should("be.visible");
      cy.findByText("foo").should("be.visible");
      cy.findByText("bar").should("be.visible");
      cy.findByText("baz").should("be.visible");
    });
```

File: e2e/test/scenarios/native-filters/helpers/e2e-date-filter-helpers.js [hunk 2]
```diff
@@ -77,5 +75,5 @@ export function setAdHocFilter({
     cy.findByText(/^Include/).click();
   }
 
-  cy.button("Add filter").click();
+  cy.button(buttonLabel).click();
```
Comment (kamilmielnik) [inline e2e/test/scenarios/native-filters/helpers/e2e-date-filter-helpers.js:78]: It'd be great if we could simplify the helpers instead of adding `buttonLabel` argument everywhere.

Here's an idea:
- make the submit button in all the popovers actually use `type="submit"` 
- make helpers find the submit button using the `type`, not the label

WDYT?

Of course we can do this in a separate PR.

**Review by kamilmielnik — APPROVED** [review]
Fix looks good :+1:

File: e2e/test/scenarios/native-filters/helpers/e2e-date-filter-helpers.js (hunk 2 above)
Comment (romeovs) [reply e2e/test/scenarios/native-filters/helpers/e2e-date-filter-helpers.js:78]: Great idea. I can set it up in a follow up idea.
